src/main/java/chess/domain/position/Direction.java

```diff
@@ -0,0 +1,117 @@
+import java.util.Arrays;
+import java.util.List;
+
+public enum Direction {
+    NORTH(0, 1),
+    NORTHEAST(1, 1),
+    EAST(1, 0),
+    SOUTHEAST(1, -1),
+    SOUTH(0, -1),
+    SOUTHWEST(-1, -1),
+    WEST(-1, 0),
+    NORTHWEST(-1, 1),
+
+    NNE(1, 2),
+    NNW(-1, 2),
+    SSE(1, -2),
+    SSW(-1, -2),
+    EEN(2, 1),
+    EES(2, -1),
+    WWN(-2, 1),
+    WWS(-2, -1);
+
+    private static final int NEXT = 1;
+
+    private final int fileGrowth;
+    private final int rankGrowth;
+
+    Direction(final int fileGrowth, final int rankGrowth) {
+        this.fileGrowth = fileGrowth;
+        this.rankGrowth = rankGrowth;
+    }
+
+    public static Direction of(final Position from, final Position to) {
+        final var fileDifference = to.getFileOrder() - from.getFileOrder();
+        final var rankDifference = to.getRankNumber() - from.getRankNumber();
+
+        return Arrays.stream(values())
+                .filter(direction -> direction.fileGrowth == fileDifference && direction.rankGrowth == rankDifference)
+                .findFirst()
+                .orElseGet(() -> getGrowthDirection(from, to));
+    }
+
+    private static Direction getGrowthDirection(final Position from, final Position to) {
+        final var fileDifference = calculateDifference(from.getFileOrder(), to.getFileOrder());
+        final var rankDifference = calculateDifference(from.getRankNumber(), to.getRankNumber());
+
+        return Arrays.stream(values())
+                .filter(direction -> direction.fileGrowth == fileDifference && direction.rankGrowth == rankDifference)
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("정의되지 않은 방향입니다."));
+    }
+
+    private static int calculateDifference(final int from, final int to) {
+        final var difference = to - from;
+        if (difference == 0) {
+            return difference;
+        }
+        return difference / Math.abs(difference);
+    }
+
+    public static List<Direction> PawnDirection(final Color color) {
+        if (color.isWhite()) {
+            return whitePawnDirection();
+        }
+        return blackPawnDirection();
+    }
+
+    public static List<Direction> linearDirection() {
+        return Arrays.asList(NORTH, EAST, SOUTH, WEST);
+    }
+
+    public static List<Direction> diagonalDirection() {
+        return Arrays.asList(NORTHEAST, SOUTHEAST, SOUTHWEST, NORTHWEST);
+    }
+
+    public static List<Direction> everyDirection() {
+        return Arrays.asList(NORTH, EAST, SOUTH, WEST, NORTHEAST, SOUTHEAST, SOUTHWEST, NORTHWEST);
+    }
+
+    public static List<Direction> knightDirection() {
+        return Arrays.asList(NNE, NNW, SSE, SSW, EEN, EES, WWN, WWS);
+    }
+
+    private static List<Direction> whitePawnDirection() {
+        return Arrays.asList(NORTH, NORTHEAST, NORTHWEST);
+    }
+
+    private static List<Direction> blackPawnDirection() {
+        return Arrays.asList(SOUTH, SOUTHEAST, SOUTHWEST);
+    }
```

Comment:
메서드로 만들 필요가 있을까요?
상수로 둬도 될 거 같아요~
```suggestion

 public static final List<Direction> LINEAR_DIRECTION = Arrays.asList(NORTH, EAST, SOUTH, WEST);
	public static final List<Direction> DIAGONAL_DIRECTION = Arrays.asList(NORTH_EAST, NORTH_WEST, SOUTH_EAST,
			SOUTH_WEST);
	public static final List<Direction> EVERY_DIRECTION = Arrays.asList(NORTH, EAST, SOUTH, WEST, NORTH_EAST,
			NORTH_WEST, SOUTH_EAST, SOUTH_WEST);
	public static final List<Direction> KNIGHT_DIRECTION = Arrays.asList(NORTH_NORTH_EAST, NORTH_NORTH_WEST,
			SOUTH_SOUTH_EAST, SOUTH_SOUTH_WEST, EAST_EAST_NORTH, EAST_EAST_SOUTH, WEST_WEST_NORTH, WEST_WEST_SOUTH);
	public static final List<Direction> WHITE_PAWN_DIRECTION = Arrays.asList(NORTH_EAST, NORTH_WEST, NORTH);
	public static final List<Direction> BLACK_PAWN_DIRECTION = Arrays.asList(SOUTH_EAST, SOUTH_WEST, SOUTH);
```